src/Interface/hiopInterface.hpp

```diff
@@ -111,109 +111,150 @@ enum hiopSolveStatus {
 {
-  /** Base class for the solver's interface that has no assumptions how the 
-   *  matrices are stored. The vectors are dense and distributed row-wise. 
-   *  The data distribution is decided by the calling code (that implements 
-   *  this interface) and specified to the optimization via 'get_vecdistrib_info'
-   *
-   *  Two possible implementations are for sparse NLPs and NLPs with small 
-   *  number of global constraints.
-   *  
-   *  
-   */
 public:
+  //Types indicating linearity or nonlinearity.
   enum NonlinearityType{ hiopLinear=0, hiopQuadratic, hiopNonlinear};
 public:
   hiopInterfaceBase() {};
   virtual ~hiopInterfaceBase() {};
 
-  /** problem dimensions: n number of variables, m number of constraints */
+  /** Specifies the problem dimensions.
+   * 
+   * @param n global number of variables
+   * @param m number of constraints
+   */
   virtual bool get_prob_sizes(long long& n, long long& m)=0;
-  /** bounds on the variables 
-   *  (xlow<=-1e20 means no lower bound, xupp>=1e20 means no upper bound) */
+  
+  /** Specifies bounds on the variables.
+   *    
+   * @param[in] n global number of constraints
+   * @param[out] xlow array of lower bounds. A value of -1e20 or less means no lower 
+   * bound is present
+   * @param[out] xupp array of upper bounds. A value of 1e20 or more means no upper 
+   * bound is present
+   * @param[out] type array of indicating whether the variables enters the objective 
+   * linearily, quadratically, or general nonlinearily. Momentarily not used by HiOp
+   */
   virtual bool get_vars_info(const long long& n, double *xlow, double* xupp, NonlinearityType* type)=0;
-  /** bounds on the constraints 
-   *  (clow<=-1e20 means no lower bound, cupp>=1e20 means no upper bound) */
+  
+  /** Specififes the bounds on the constraints.
+   *
+   * @param[in] m number of constraints
+   * @param[out] clow array of lower bounds for constraints. A value of -1e20 or less means no lower 
+   * bound is present
+   * @param[out] cupp array of upper bounds for constraints. A value of 1e20 or more means no upper 
+   * bound is present
+   * @param[out] type array of indicating whether the constraint is linear, quadratic, or general
+   * nonlinear. Momentarily not used by HiOp
+   */
   virtual bool get_cons_info(const long long& m, double* clow, double* cupp, NonlinearityType* type)=0;
 
-  /** Objective function evaluation
-   *  When MPI enabled, each rank returns the obj. value. Also, x points to the local entries and 
-   *  the function is responsible for knowing the local buffer size.
+  /** Method the evaluation of the objective function.
+   *  
+   * @param[in] n global size of the problem
+   * @param[in] x array with the local entries of the primal variable
+   * @param[in] new_x whether x has been changed from the previous calls to other evaluation methods
+   * (gradient, constraints, Jacobian, and Hessian),
+   * @param[out] obj_value the value of the objective function at @p x
```

Comment:
good to know but 
- md does not cover all the features of doxygen (e.g.,  `@param`)
- it will be inconsistent to use md when the class is documented with doxygen javadoc style.